procedures/unit-testing-test-md-gen.ipf

```diff
@@ -191,57 +220,132 @@ static Function EvaluateDgenTagResult(err, fullFuncName, varName)
 	if(!WaveExists(wGenerator))
-		sprintf msg, "Data Generator function %s returns a null wave. It is referenced by test case %s.", dgen, fullFuncName
-		UTF_Reporting#ReportErrorAndAbort(msg)
-	elseif(DimSize(wGenerator, UTF_COLUMN) > 0)
-		sprintf msg, "Data Generator function %s returns not a 1D wave. It is referenced by test case %s.", dgen, fullFuncName
+		// wave is not stored
+		sprintf msg, "Data Generator function \"%s\" has no suitable data for test case \"%s\".", dgen, fullFuncName
 		UTF_Reporting#ReportErrorAndAbort(msg)
 	elseif(!((wType1 == IUTF_WAVETYPE1_NUM && WaveType(wGenerator, 1) == wType1 && WaveType(wGenerator) & wType0) || (wType1 != IUTF_WAVETYPE1_NUM && WaveType(wGenerator, 1) == wType1)))
-		sprintf msg, "Data Generator %s functions returned wave format does not fit to expected test case parameter. It is referenced by test case %s.", dgen, fullFuncName
+		sprintf msg, "Data Generator \"%s\" functions returned wave format does not fit to expected test case parameter. It is referenced by test case \"%s\".", dgen, fullFuncName
 		UTF_Reporting#ReportErrorAndAbort(msg)
 	elseif(!UTF_Utils#IsNaN(wRefSubType) && wType1 == IUTF_WAVETYPE1_WREF && !UTF_Utils#HasConstantWaveTypes(wGenerator, wRefSubType))
-		sprintf msg, "Test case %s expects specific wave type1 %u from the Data Generator %s. The wave type from the data generator does not fit to expected wave type.", fullFuncName, wRefSubType, dgen
+		sprintf msg, "Test case \"%s\" expects specific wave type1 %u from the Data Generator \"%s\". The wave type from the data generator does not fit to expected wave type.", fullFuncName, wRefSubType, dgen
 		UTF_Reporting#ReportErrorAndAbort(msg)
 	endif
 
 	return wGenerator
 End
+
+static Function ExecuteAllDataGenerators(debugMode)
+	variable debugMode
+
+	variable i, j, size, length, dimPos, err
+	string dgenList, dgen, fullFuncName, endTime, msg, procWin
+	variable DGENLIST_Index
+
+	WAVE/T testRunData = UTF_Basics#GetTestRunData()
+	size = DimSize(testRunData, UTF_ROW)
+	DGENLIST_Index = FindDimLabel(testRunData, UTF_COLUMN, "DGENLIST")
+
+	if(!size)
+		return NaN
+	endif
+
+	WAVE/WAVE wDgen = GetDataGeneratorWaves()
+
+	UTF_Reporting_Control#TestSuiteBegin("@DGEN_SUITE")
+
+	for(i = 0; i < size; i += 1)
+		dgenList = testRunData[i][DGENLIST_Index]
+		WAVE/T dgenWave = ListToTextWave(dgenList, ";")
+		length = DimSize(dgenWave, UTF_ROW)
+
+		for(j = 0; j < length; j += 1)
+			dgen = dgenWave[j]
+			dimPos = FindDimLabel(wDgen, UTF_ROW, dgen)
+			if(dimPos != -2)
+				continue
+			endif
+			procWin = StringByKey("PROCWIN", FunctionInfo(dgen))
+
+			FUNCREF TEST_CASE_PROTO_DGEN fDgen = $dgen
+			if(!UTF_FuncRefIsAssigned(FuncRefInfo(fDgen)))
+				fullFuncName = testRunData[i][%FULLFUNCNAME]
+				sprintf msg, "Data Generator function \"%s\" has wrong format. It is referenced by test case \"%s\".", dgen, fullFuncName
+				UTF_Reporting#ReportErrorAndAbort(msg)
+			endif
+
+			UTF_Debug#SetDebugger(debugMode)
```

Comment:
Setting the Debugger at runtime is rather slow. For the regular TCs we do that is TestBegin/TestEnd.
Please don't set the Debugger for each DGEN-TC, it is sufficient to set/restore it outside the for loops here.

Reply:
Thanks, I fixed it.